### PR TITLE
Add an option to support creating automatic APIreview from manual pipe…

### PR DESCRIPTION
**Pull request by praveenkuttappan**
…line run only

This PR is to support python azure-ml team to create automatic API review only from release pipeline instead of creating it by scheduled run since scheduled run always create API review from main branch and manual one creates from the branch it's triggered against.

### CODE REVIEW

**Comment by praveenkuttappan** [comment]
This PR will disable automatic API review gen from main branch by scheduled CI run for azure-ml package and APi review will get created only when running pipeline manually. This should address main concern of new API view revision getting removed by CI for azure-ml.

**Review by scbedd — APPROVED** [review]
Honestly the yml comment doesn't bring very much clarity to my slow brain. After mapping this out in a truth table I agree with your choices though.

```
BuildReason = A = manually queued
ShouldSkip = B = Parameter set to false

A     | B     | Includes In Yaml
True  | True  | True
False | False | False
True  | False | True
False | True  | True
```

**Comment by weshaggard** [comment]
Perhaps I'm missing some context but can why do we want to disable automatic review but enable reviews for manual runs? That doesn't seem to fit with our general model. I could understand potentially having an opt-out for the automatic reviews but I would not automatically opt-in to it for all manual runs. I would instead potentially have folks have a queue time variable they could set to override this as needed.

**Comment by praveenkuttappan** [comment]
@weshaggard I have updated PR as we discussed to correct flag name and change the condition for clarify.

**Comment by praveenkuttappan** [comment]
If flag `GenerateApiReviewForManualOnly` is `false` (default value): APi review is generated for all type of pipeline runs.
If flag `GenerateApiReviewForManualOnly` is `true`: API review is generated only for manual pipeline runs.

File: eng/pipelines/templates/steps/create-apireview.yml [hunk 1]
```diff
@@ -4,34 +4,39 @@ parameters:
...
-      and(
-        succeededOrFailed(),
-        ne(variables['Skip.CreateApiReview'], 'true'),
-        ne(variables['Build.Reason'],'PullRequest'),
-        eq(variables['System.TeamProject'], 'internal'),
-        not(endsWith(variables['Build.Repository.Name'], '-pr'))
-      )
+ 
+  # Automatic API review is generated for a package when pipeline runs irrespective of how pipeline gets triggered.
+  # One excpetion to this case is Python Azure ML package. Azure ML does not want API reviews to be created by scheduled and individual CI runs.
```
Comment (weshaggard) [inline eng/pipelines/templates/steps/create-apireview.yml:15]: I would remove the ml mention here as we have that context in the ml ci.yml file.

File: sdk/ml/ci.yml [hunk 2]
```diff
@@ -30,6 +30,9 @@ extends:
     ServiceDirectory: ml
     ValidateFormatting: true
     TestProxy: true
+    # This is a short term solution to create API review for python azure-ml package only when running pipeline manually
+    # Long term solution should be to have different versions on main branch and release branch for python package so APIView can have different revisions for each version.
```
Comment (weshaggard) [inline sdk/ml/ci.yml:34]: Can we please file an issue for this and add the link in this comment?

**Review by weshaggard — APPROVED** [review]
Left a couple clean-up comments but otherwise looks fine.

**Pull request merged**